src/Workspaces/Core/Portable/Workspace/Workspace_Events.cs

```diff
@@ -160,7 +160,7 @@ protected Task RaiseDocumentClosedEventAsync(Document document)
         /// An event that is fired when the active context document associated with a buffer 
         /// changes.
         /// </summary>
-        internal event EventHandler<DocumentEventArgs> DocumentActiveContextChanged
+        public event EventHandler<DocumentEventArgs> DocumentActiveContextChanged
```

Comment:
I don't know if i'm a fan of this sending the Document along as an event arg.  It's not clear what that Document is.  Is it the old Active Document?  THe new one?  I don't like that is also keeps the entire Solution alive.  I think i'd prefer if this just exposed a DocumentId instead.